### PR TITLE
yum repo utils: detect if the repourl already has an md5sum

### PR DESCRIPTION
**Pull request by mlangsdorf**
Don't append md5sums to repourls that already have md5sums.

Signed-off-by: Mark Langsdorf <mlangsdo@redhat.com>

### CODE REVIEW

File: atomic_reactor/yum_util.py [hunk 1]
```diff
@@ -54,7 +54,12 @@ def filename(self):
         '''
         urlpath = unquote(urlsplit(self.repourl, allow_fragments=False).path)
         basename = os.path.basename(urlpath)
-        suffix = '-' + md5(self.repourl.encode('utf-8')).hexdigest()[:5]
+        poss_name = self.repourl[:-11] + REPO_SUFFIX
+        # check if this name is already encoded
+        if md5(poss_name.encode('utf-8')).hexdigest()[:5] == self.repourl[-10:-5]:
```
Comment (lcarva) [inline atomic_reactor/yum_util.py]: Do we know what's causing it to be encoded twice?

**Comment by twaugh** [comment]
We shouldn't have to be guessing what's in the filename. This looks like, to avoid problems when hashing twice, we hash a third time. Can we avoid the second hash instead?

**Comment by mlangsdorf** [comment]
if I'm following the code:
* pre_add_yum_repo_by_url, pre_koji, and pre_flatpak_create_dockerfile all add yum repos to workflow.files.
* pre_inject_yum_repo consumes workflow.files by writing the repos' content out to a local directory and adjusting the Dockerfile.

I could add a flag to YumRepo that controls the hash, so pre_inject_yum_repo wouldn't rehash.

File: atomic_reactor/yum_util.py (hunk 1 above)
Comment (mlangsdorf) [reply atomic_reactor/yum_util.py]: another approach

File: atomic_reactor/plugins/pre_add_yum_repo.py [hunk 2]
```diff
@@ -33,7 +33,8 @@ def __init__(self, tasker, workflow, repo_name, baseurl):
         """
         # call parent constructor
         super(AddYumRepoPlugin, self).__init__(tasker, workflow)
-        self.repo = YumRepo(baseurl=baseurl, name=repo_name, enabled=True)
+        self.repo_name = repo_name
```
Comment (rcerven) [inline atomic_reactor/plugins/pre_add_yum_repo.py]: those 2 new lines are missing coverage

Comment (lcarva) [reply atomic_reactor/plugins/pre_add_yum_repo.py]: This whole class is not tested :disappointed: 
Unrelated to this change though

**Comment by lcarva** [comment]
Not updating release notes draft because this is bugfix to an unreleased feature.

**Pull request merged**